pkg/virtctl/imageupload/imageupload.go

```diff
@@ -926,3 +935,38 @@ func handleEventErrors(client kubecli.KubevirtClient, pvcName, dvName, namespace
 
 	return nil
 }
+
+func createDataSource(client kubecli.KubevirtClient, name, namespace string) error {
+	if ds, _ := client.CdiClient().CdiV1beta1().DataSources(namespace).Get(context.Background(), name, metav1.GetOptions{}); ds != nil {
```

Comment:
Use [`IsNotFound`](https://pkg.go.dev/k8s.io/apimachinery/pkg/api/errors#IsNotFound) to check if the resource was not found. Otherwise please return the error.

Reply:
If the resource was not found, shouldn't we create a new one then?

Reply:
Okay I've refactored it. I think it looks better.